File: src/test/java/com/lsd/events/NewpageTest.java
package com.lsd.events;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class NewpageTest {

    @Test
    void produceExpectedMarkup() {
        var markup = Newpage.titled("my title").toMarkup();

        assertThat(markup).isEqualTo("\nnewpage 'my title'\n");
    }

    @Test
    void handleEmptyTitle() {
        var markup = Newpage.titled("").toMarkup();

        assertThat(markup).isEqualTo("\nnewpage\n");
    }
    
    @Test
    void handleNullTitle() {
        var markup = Newpage.titled(null).toMarkup();

        assertThat(markup).isEqualTo("\nnewpage\n");
    }
}

File: src/test/kotlin/com/lsd/events/NewpageTest.kt
package com.lsd.events

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class NewpageTest {
    
    @Test
    fun produceExpectedMarkup() {
        assertThat(Newpage.titled("my title").toMarkup())
            .isEqualTo("\nnewpage 'my title'\n")
    }

    @Test
    fun handleEmptyTitle() {
        assertThat(Newpage.titled("").toMarkup())
            .isEqualTo("\nnewpage\n")
    }
}